avm/res/network/network-manager/connectivity-configuration/main.bicep

```diff
@@ -66,3 +66,29 @@ output resourceId string = connectivityConfiguration.id
 
 @sys.description('The resource group the connectivity configuration was deployed into.')
 output resourceGroupName string = resourceGroup().name
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type appliesToGroupsType = {
+  @sys.description('Required. Group connectivity type.')
+  groupConnectivity: ('DirectlyConnected' | 'None')
+
+  @sys.description('Optional. Flag if global is supported.')
+  isGlobal: bool?
+
+  @sys.description('Required. Network group Id.')
+  networkGroupId: string
+
+  @sys.description('Optional. Flag if use hub gateway.')
+  useHubGateway: bool?
+}[]
+
+type hubsType = {
+  @sys.description('Required. Resource Id of the hub.')
+  resourceId: string
+
+  @sys.description('Required. Resource type of the hub.')
+  resourceType: 'Microsoft.Network/virtualNetworks'
```

Comment:
Do understand it right that this is the only allowed type? If so, would it not make sense to just make it optional and set it to set value by default?

Reply:
Currently yes but given how its implemented. Appears like potentially supporting something else down the track like virtual wan. See [documentation](https://learn.microsoft.com/en-us/azure/virtual-network-manager/faq#can-an-azure-virtual-wan-hub-be-part-of-a-network-group).